src/lib.rs

```diff
@@ -180,8 +197,21 @@ impl<T> StableRegion<T> {
             // We could choose a different rule here if we wanted to be
             // more conservative with memory (e.g. page size allocations).
             let next_len = (self.local.capacity() + 1).next_power_of_two();
-            let new_local = Vec::with_capacity(std::cmp::max(count, next_len));
+            let capacity = std::cmp::max(count, next_len);
+            let capacity = std::cmp::max(capacity, 1<<12 / std::mem::size_of::<T>());
+            let file = tempfile::tempfile().unwrap();
```

Comment:
Worth being clear that tempfile's tmpdir might point to a tmpfs, which would make these allocations effectively anonymous instead of file-backed, which would be an operational gotcha if the intent is to allow the OS to page out for us.